SemiconductorTestLibrary.Extensions/tests/Unit/InstrumentAbstraction/DAQmx/AnalogInputTests.cs

```diff
@@ -56,6 +62,32 @@ public void ReadFiveAnalogSamplesFromTwoChannels_ResultsContainExpectedData()
             Assert.Equal(5, results.ExtractSite(1)["VCC2"].Length);
         }
 
+        [Fact]
+        [Trait(nameof(Platform), nameof(Platform.TesterOnly))]
+        [Trait(nameof(HardwareConfiguration), nameof(HardwareConfiguration.Lungyuan))]
+        public void InitializeDAQmxTasks_ReadAnalogSamplesFromOneFilteredChannel_ResultsContainExpectedData()
+        {
+            var sessionManager = Initialize("DAQmxMultiChannelTests.pinmap");
+            InitializeAndClose.CreateDAQmxAOVoltageTasks(_tsmContext);
+            var aiTasksBundle = sessionManager.DAQmx("AIPin");
+            var aoTasksBundle = sessionManager.DAQmx("AOPin");
+            var data = new PinSiteData<double>(new Dictionary<string, IDictionary<int, double>> { { "AOPin", new Dictionary<int, double> { { 0, 0.5 }, { 1, 0.8 } } } });
+            ConfigureTimingSampleClockSettings(aiTasksBundle, aoTasksBundle);
+
+            aoTasksBundle.WriteAnalogSingleSample(data, autoStart: false);
+            aiTasksBundle.Start();
+            aoTasksBundle.Start();
+            // Wait for the generation to get completed
+            Thread.Sleep(100);
+            aoTasksBundle.Stop();
+            var filteredSiteData = aiTasksBundle.FilterBySite(1).ReadAnalogMultiSample();
+            aiTasksBundle.Stop();
+            var maxValueOfFilteredSamples = filteredSiteData.GetValue(filteredSiteData.SiteNumbers[0], "AIPin").Max();
+
+            AssertFilteredSample(maxValueOfFilteredSamples, aiTasksBundle, 0.45, 0.5);
```

Comment:
```suggestion
            AssertFilteredSample(maxValueOfFilteredSamples, aiTasksBundle, 0.75, 0.8);
```

If filtered for site1 you should assert for 0.8 since that's the value we set for Site1.

Reply:
Updated